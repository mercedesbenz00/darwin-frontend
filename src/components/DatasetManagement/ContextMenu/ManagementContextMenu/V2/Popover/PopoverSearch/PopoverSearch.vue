<template>
  <div class="popover-search">
    <input-field-v2
      v-bind="$attrs"
      v-on="$listeners"
    >
      <template #left-icon>
        <slot name="left-icon" />
      </template>
      <template #right-icon>
        <slot name="right-icon" />
      </template>
    </input-field-v2>
  </div>
</template>

<script lang='ts'>
import { defineComponent } from 'vue'

import InputFieldV2 from '@/components/Common/InputField/V2/InputField.vue'

/**
 * @Component PopoverSearch
 * ~ Reuseable component for the Popover. Basically just a individual styled wrapper with the input
 * in it.
 * */

export default defineComponent({
  name: 'PopoverSearch',
  components: { InputFieldV2 },
  setup (props) {
    return {
      props
    }
  }
})
</script>

<style lang='scss' scoped>
.popover-search {
  position: sticky;
  bottom: 0px;
  left: 0px;

  display: block;
  padding: 6px;
  background: $colorSurfaceElevate;
  box-shadow: $borderLightTop;

  width: 100%;
}
</style>
