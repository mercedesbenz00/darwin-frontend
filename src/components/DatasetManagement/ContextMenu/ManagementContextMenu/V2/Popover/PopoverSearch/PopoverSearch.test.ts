import { shallowMount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'

import PopoverSearch from './PopoverSearch.vue'

let wrapper: Wrapper<Vue>

const propsData = {
  value: ''
}

beforeEach(() => {
  wrapper = shallowMount(PopoverSearch, { propsData })
})

afterEach(() => {
  wrapper.destroy()
})

it('should render properly', () => {
  expect(wrapper.exists()).toBeTruthy()
})

it('should match snapshot', () => {
  expect(wrapper).toMatchSnapshot()
})
