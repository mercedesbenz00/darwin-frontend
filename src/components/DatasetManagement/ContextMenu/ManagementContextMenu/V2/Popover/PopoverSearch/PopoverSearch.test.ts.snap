// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div class="popover-search" value="">
  <input-field-v2-stub value="" tag="div" type="text" items=""></input-field-v2-stub>
</div>
`;
